flink-autoscaler/src/main/java/org/apache/flink/autoscaler/exceptions/RecoverableException.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.autoscaler.exceptions;
+
+/** An exception for a recoverable scenario. */
+public class RecoverableException extends RuntimeException {
```

Comment:
How about rename it to `IgnorableException` or `NotReadyException`?

Recoverable generally means that it can be recovered and will be tried again later. `JobAutoScalerImpl#scale` doesn't retry it, and just doesn't report `onError`. And regardless of whether it is a RecoverableException this time, the next control loop will perform the same logic.

Based on it, I think the `RecoverableException` isn't very accurate here. WDYT?

Reply:
Fair. At one point I actually had `NotReadyException` :) Renamed to `NotReadyException`.